mne/time_frequency/tfr.py

```diff
@@ -1107,9 +1162,245 @@ def plot(self, picks, baseline=None, mode='mean', tmin=None, tmax=None,
+            Defaults to ``None``.
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see :func:`mne.verbose`).
+
+        Returns
+        -------
+        fig : matplotlib.figure.Figure
+            The figure containing the topography.
+        """  # noqa: E501
+        from ..viz.topomap import _set_contour_locator
+        from ..viz.utils import _setup_vmin_vmax
+        import matplotlib.pyplot as plt
+
+        if topomap_args is None:
+            topomap_args = dict()
+
+        cmap = _setup_cmap(cmap)
+
+        fig = plt.figure()
+        tf_ax = fig.add_subplot(212)
+
+        _, tf_data, tf_times, tf_freqs = \
+                             self._plot(picks=picks, baseline=baseline,
+                             mode=mode,tmin=tmin, tmax=tmax, fmin=fmin,
+                             fmax=fmax, vmin=vmin, vmax=vmax, cmap=cmap,
+                             dB=dB, colorbar=False, show=show, title=title,
+                             axes=tf_ax, layout=layout, yscale=yscale,
+                             aggregate=aggregate, exclude=exclude,
+                             verbose=verbose)
+
+        vmin, vmax = _setup_vmin_vmax(tf_data, vmin, vmax)
+
+        if timefreqs is None:
+            # find the maximum peak
+            from scipy.signal import argrelmax
+            order = tf_data.shape[2] // 30
+            if order < 1:
+                order = 1
+            peaks_idx = argrelmax(tf_data, order=order, axis=2)
+            if peaks_idx[0].size == 0:
+                _,p_t,p_f = np.unravel_index(tf_data.argmax(), tf_data.shape)
+                timefreqs = [(tf_times[p_t], tf_freqs[p_f])]
+            else:
+                peaks = [tf_data[0,f,t] for f,t in zip(peaks_idx[1], peaks_idx[2])]
+                peakmax_idx = np.argmax(peaks)
+                peakmax_time = tf_times[peaks_idx[2][peakmax_idx]]
+                peakmax_freq = tf_freqs[peaks_idx[1][peakmax_idx]]
+
+                timefreqs = [(peakmax_time, peakmax_freq)]
+
+        # prepare axes for topomap
+        # slightly convoluted due to colorbar placement and for vertical alignment
+        ts = len(timefreqs) + 2
+        map_ax = [plt.subplot(4, ts, x + 2 + ts) for x in range(ts - 2)]
+        cbar_ax = plt.subplot(4, 3 * (ts + 1), 6 * (ts + 1))
+
+        # topomap
+        contours = topomap_args.get('contours', 6)
+        locator, contours = _set_contour_locator(vmin, vmax, contours)
+
+        topomap_args_pass = dict((k, v) for k, v in topomap_args.items() if
+                                 k not in ['axes', 'show', 'colorbar'])
+        topomap_args_pass['outlines'] = (topomap_args['outlines'] if 'outlines'
+                                         in topomap_args else 'skirt')
+        topomap_args_pass['contours'] = contours
+
+        for idx in range(len(map_ax)):
+            time = timefreqs[idx][0]
+            freq = timefreqs[idx][1]
```

Comment:
@jona-sassenhagen we also need `idx` for getting the average parameters (`timefreqs_average`) in the case where it is a list. That would give something like
```
for idx, ((time, freq), ax) in enumerate(zip(timefreqs, map_ax)):
    ...
```
I switched to a simple ` for idx in range(len(map_ax)):` because it was getting a bit cramped up. Do you think it's still better to follow the pythonic way?